podvm/files/etc/systemd/system/kata-opa.service

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2023 Microsoft Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+[Unit]
+Description=Open Policy Agent for Kata Containers
+Documentation=https://github.com/kata-containers
+ConditionPathExists=/etc/kata-opa/default-policy.rego
+
+# kata-agent connects to OPA while starting up.
+Before=kata-agent.service
```

Comment:
I don't think this reliably guarantees that opa has finished launching before kata, I think we want to have a `Requires` + `After` on the kata unit file, or we add a `RequiredBy=kata-agent.service` here. Personally I find the first option more intuitive

Reply:
I took the configs from kata main. 
Ref: https://github.com/kata-containers/kata-containers/blob/main/src/kata-opa/kata-opa.service.in

I can open an issue in main as well to make the changes.

Reply:
https://www.digitalocean.com/community/tutorials/understanding-systemd-units-and-unit-files#unit-section-directives

> Before=: The units listed in this directive will not be started until the current unit is marked as started if they are activated at the same time. This does not imply a dependency relationship and must be used in conjunction with one of the above directives if this is desired

Reply:
Looks like there is no need to modify the unit files as the agent retries connecting to the opa client - https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/src/policy.rs#L91

Do you still suggest changing the unit files for either the kata-agent or kata-opa service ?

Reply:
IMO, it's not particularly pretty to have failed attempts in the log file. if we allow OPA to be disabled, we could add a "RequiredBy=" directive to the opa unit, so the kata unit will fail if OPA fails to start.